discovery/api/v1/client/http.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2024 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/nuts-foundation/go-did/vc"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/discovery/api/v1/model"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// New creates a new HTTPInvoker.
+func New(strictMode bool, timeout time.Duration, tlsConfig *tls.Config) *HTTPInvoker {
+	return &HTTPInvoker{
+		client: core.NewStrictHTTPClient(strictMode, timeout, tlsConfig),
+	}
+}
+
+var _ Invoker = &HTTPInvoker{}
```

Comment:
I wanted to have an interface, which shouldn't be named `Client` since it's already in a package named `client`. Naming it `HTTPClient` makes naming the implementation hard, since it would be something like `DefaultHTTPClient`. But I guess that's the least of all evils, although it includes the package name. But consistency is probably the most important here (again).